framework/scripts/framework_version.sh

```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+git log -n1 --pretty=format:'Commit: %H%nAuthor: %aN <%ae>%n%n%B%n'
```

Comment:
I don't think this would work if someone downloads the test framework as a .tar.gz file from the releases section of GitHub. And this may fail then. 

I'm just thinking out possibilities here.. If we can have a git.properties file similar to Drill, it may help?

Reply:
[Git commit id plugin in Drill](https://github.com/apache/drill/blob/master/pom.xml#L539) is what @Agirish is talking about. We could try it out.

Reply:
Need some time to try it out.

Reply:
@Agirish, @aravi5, 
As it appears, in cases when .git folder is absent git commit plugin also won't work. Although I can do this, and it appears to me more flexible than running a command from console.

Reply:
@agozhiy, I see. But we expect the git commit plugin to be only run when building the framework - which is expected to be via git clone.

So there are two ways people can use the drill-test-framework:
(1) git clone - build and then run tests. Here the git plugin approach should work.
(2) use a tar ball of the framework. Here, the expectation is that the tarball already contains the git.properties file. When building the tar ball, it's expected to be via step (1). 

On GitHub, I've been doing periodic releases (coinciding with Drill versions). But I think that just includes the src - not a built version. We should find a space to build and host the tar ball releases.

Reply:
Also, the git commit plugin approach appears to be more standard, so I think it would be good to go that route than running a command.

Reply:
Update: Looks like we can host the built tar-ball on GitHub under releases, along with the source. They need to be uploaded. Starting next release, we can do that.

Reply:
I agree.
And as I see, we need Drill jdbc driver to build framework, wouldn't it be a problem for creating tar-balls?

Reply:
Actually for that, I'm thinking by default we should just pick up the latest release version of Apache Drill in the pom dependency - rather than using the system path we have today. 

But we should have an option to override it in our Jenkins test env, as we also want to test the latest JDBC Driver. This is mainly to make sure any changes in JDBC layer are also caught. Plus we historically had issues with compatibility across versions of Drill - which is a lot better today.

This way anyone can try out the test framework without having a runtime dependency. And for most tests that should be sufficient.